src/app/shared/components/category-navigation/category-navigation.component.html

```diff
@@ -5,6 +5,7 @@
         *ngIf="!category.disabled"
         class="category-link"
         [routerLink]="['/', outputPath, routingCategoryName[category.name]]"
+        [queryParams]="queryParams()"
```

Comment:
Try to listen to queryParams chenges and the assign it to queryParams$  and invoke async pipe.

Reply:
I'm not sure I can listen to it unfortunately. The value comes from localstorage. It's not an observable or promise.

Reply:
I think it may be more difficult when we have lazy loading, maybe reacting to router events could help here

Reply:
Do you think there may be issues with current approach? I can play with navigation events, but not sure if it's really worth it.

Reply:
We had a discussion at today's daily that the localStorage-based approach has many issues like causing problems for multiple tabs open and that it'd be better to keep data in router state instead. I think @maciejp-github is working on this?

Reply:
Great! 
In this case there will be change in `getDefaultQueryParams` function.

If I try to listen to the router events here, I'll probably do sth conflicting with Maciej :)

We can either wait with this PR for the change from Maciej, or merge it now and let Maciej handle it in scope of removing localstorage. 

Not sure what's better.

Reply:
I talked to Maciek, we're going to merge this now.